slp2mp4/dolphinrunner.py

```diff
@@ -0,0 +1,108 @@
+import os, sys, subprocess, time, shutil, uuid, json
+
+MAX_WAIT_SECONDS = 8 * 60 + 10
+
+class DolphinRunner:
+
+    def __init__(self, conf, user_dir, working_dir, job_id):
+        self.conf = conf
+        self.user_dir = user_dir
+        self.working_dir = working_dir
+        self.job_id = job_id
+
+        # get all needed paths
+        self.comm_file = os.path.join(self.working_dir, 'slippi-comm-{}.txt'.format(self.job_id))
+        self.render_time_file = os.path.join(self.user_dir, 'Logs', 'render_time.txt')
+        self.dump_dir = os.path.join(self.user_dir, 'Dump')
+        self.frames_dir = os.path.join(self.dump_dir, 'Frames')
+        self.audio_dir = os.path.join(self.dump_dir,'Audio')
+        self.video_file0 = os.path.join(self.frames_dir, 'framedump0.avi')
+        self.video_file1 = os.path.join(self.frames_dir, 'framedump1.avi')
+        self.audio_file = os.path.join(self.audio_dir, 'dspdump.wav')
+
+    def count_frames_completed(self):
+        num_completed = 0
+
+        if os.path.exists(self.render_time_file):
+            with open(self.render_time_file, 'r') as f:
+                num_completed = len(list(f))
+
+        print("Rendered ",num_completed," frames")
+        return num_completed
+
+    def run(self, slp_file, num_frames):
```

Comment:
Good job breaking up most of the code but I think this function could be broken up more. Comments are helpful but each block of code separated by a comment should really be its own function in this class.